### PR TITLE
adds nofollow and blank target

### LINKED ISSUE #16636: Open page in new tab option
Implement the option to open any menu item in a new tab and add the option to the inspector.

### PR DESCRIPTION
**Pull request by draganescu**
## Description
Closes #16636 

## How has this been tested?
Tested locally.

## Types of changes
Added two functions that handle the rel and target attributes accordingly. I used classnames as a tokenizer for lack of a better option, but I believe classnames works for `rel` just as well as for `class`.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: packages/block-library/src/navigation-menu-item/block.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
 		"destination": {
 			"type": "string"
 		},
+		"rel": {
```
Comment (jorgefilipecosta) [inline packages/block-library/src/navigation-menu-item/block.json]: It seems the (block) attribute rel can be computed from the attributes opensInNewTab and nofollow. If that's the case can we remove the rel attribute, and on the save function depending on the attributes opensInNewTab and nofollow we save the current value for the rel (HTML attribute)?

Comment (draganescu) [reply packages/block-library/src/navigation-menu-item/block.json]: yes, but why? Isn't rel an attribute anyway? Is there a reason to have less attributes, especially when we are actually printing attributes on render?

**Comment by paaljoachim** [comment]
Similar PR here: https://github.com/WordPress/gutenberg/pull/16609

File: packages/block-library/src/navigation-menu-item/block.json (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/block-library/src/navigation-menu-item/block.json]: The reason is that each time we update the original attribute we also need to update the derived attribute forgetting to do that introduces bugs. In the future, one may easily add a logic that changes nofollow attribute and forgets to change the rel attribute. Another reason is that saving an attribute that can be computed adds more information in the database without a need.

**Comment by draganescu** [comment]
Closing, to be treated in a wider perspective.

**Pull request closed without merging**